scipy/stats/_entropy.py

```diff
@@ -144,8 +144,21 @@ def entropy(pk: np.typing.ArrayLike,
         S /= np.log(base)
     return S
 
+
+def _differential_entropy_is_too_small(samples, kwargs):
+    values = samples[0]
+    n = values.shape[-1]
+    window_length = kwargs.get("window_length",
+                               math.floor(math.sqrt(n) + 0.5))
+    if not 2 <= 2 * window_length < n:
+        return True
+    return False
+
+
 @_axis_nan_policy_factory(
-    lambda x: x, n_outputs=1, result_to_tuple=lambda x: (x,)
+    lambda x: x, n_outputs=1, result_to_tuple=lambda x: (x,),
+    too_small=_differential_entropy_is_too_small,
+    override={"nan_propagation": False}
```

Comment:
Wait, what happens without this? I would prefer for the decorator to override here. I thought all we needed to do was override `too_small`?

Reply:
I kept this to just preserve backward compatibility while we decide if we want to raise a ValueError here or return a NaN.

Reply:
Do any tests fail?

Reply:
Nope, just that now instead of returning a NaN for `nan_policy="propagate"`, the function will throw a `ValueError` if any input slice is too small.

Reply:
Just to be clear - the only difference in behavior you're aware of is that when:

- `nan_policy='propagate'`
- the array is 1d
- the array is too small

The decorator would return a NaN, but the function currently (and when we don't let the decorator override) raises a ValueError?

Reply:
> Just to be clear - the only difference in behavior you're aware of is that when:
> 
>     * `nan_policy='propagate'`
> 
>     * the array is 1d
> 
>     * the array is too small

The array doesn't have to be 1D. For nD arrays, the function would raise a ValueError when any slice is too small. For example,

```python
from scipy import stats

stats.differential_entropy([[np.nan]*5, [1,2,3,4,5]])  # ValueError
```

> Either way passes tests?

Yes.

Reply:
Meh, I'm not really concerned. This is fine, and as you say, it better preserves the original behavior.
Someday, I want to stop raising the errors with 1d arrays and return NaN instead to keep the behavior consistent whether it's 1D or Nd. But that day is not today.

Reply:
> For nD arrays, the function would raise a ValueError when any slice is too small. For example,

Reply:
Reverted this change in #19346